### DISCUSSION TITLE
[Script Request] Docmost

### DISCUSSION THREAD

**Discussion by rlkollman** [community-scripts/ProxmoxVE#785] [Request script]
### Application Name

Docmost

### Website

https://github.com/docmost/docmost

### Description

Docmost is an open-source collaborative wiki and documentation software that allows you to create and share knowledge seamlessly and is ideal for managing your wiki, knowledge-base, documentation and a lot more.

I did see someone recently requested this, however, not utilizing the Request Script method, so I am posting it.  I know there are a number of similar apps within Community-Scripts, but I find the layout and appearance of this one to be more appealing.  I think the appearance of the content is more centered and focal to the user viewing experience, so users can see the materials without a lot of big blocks of wasted space.  I like a wiki that shows you what you came to see and I think Docmost does it well.  The real-time collaborative editor is also a perk for active groups.

Thank you :)

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by JesseWebDotCom** [comment]
I just created a script for this (see attached - just download and remove the .md extension).  I'm only minutes into testing but I can create a create a user, login, create a workspace, create a page, etc. The only thing I couldn't figure out is container automatic root login (so I just use pct enter <container-id> to login to the container console).
[docmost.sh.md](https://github.com/user-attachments/files/18222749/docmost.sh.md)

**Reply by JesseWebDotCom** [reply]
Configuration settings (APP_URL, email, etc.) work:

- https://docmost.com/docs/self-hosting/configuration/
- https://docmost.com/docs/self-hosting/environment-variables

Simply do this after install:

1. nano /opt/docmost/.env
2. Add your variables (ex. APP_URL=https://your_docmost_url), save, and exit nano
3. systemctl restart docmost

**Reply by jaygz316** [reply]
Awesome, I've been looking for Docmost, I'll try it out right now. Thank you

**Reply by rlkollman** [reply]
Thank you for taking the time to do this!  It did fail installing for me.  I ran the installer twice with verbosity, and in both instances, it looks like the `nx run server:build` failed.

The settings I used (settings I changed are **bold**) and I'm running Proxmox 8.3.0:
  🖥️  Operating System: **debian**
  🌟  Version: **12**
  📦  Container Type: Unprivileged
  🔐  Root Password: ****
  🆔  Container ID: **333**
  🏠  Hostname: docmost
  💾  Disk Size: **10**
  🧠  CPU Cores: 2
  🛠️  RAM Size: 4096
  🌉  Bridge: vmbr0
  📡  IP Address: dhcp
  🌐  Gateway IP Address: Default
  📡  APT-Cacher IP Address: Default
  🚫  Disable IPv6: no
  ⚙️  Interface MTU Size: Default
  🔍  DNS Search Domain: Host
  📡  DNS Server IP Address: Host
  🏷️  Vlan: Default
  🔑  Root SSH Access: **yes**
  🔍  Verbose Mode: yes

> ✖ nx run server:build
> server@0.6.2 build /opt/docmost/apps/server
> nest build
>      src/collaboration/collaboration.util.ts:34:8 - error TS2307: Cannot find module '@docmost/editor-ext' or its corresponding type declarations.
>      34 } from '@docmost/editor-ext';
>                ~~~~~~~~~~~~~~~~~~~~~
>      Found 1 error(s).
>       ELIFECYCLE  Command failed with exit code 1.

> NX   Ran target build for 3 projects (1m)
> ✔  2/3 succeeded [0 read from cache]
>   ✖  1/3 targets failed, including the following:
>      - nx run server:build
> ⠏ ELIFECYCLE  Command failed with exit code 1.
> [ERROR] in line 76: exit code 0: while executing command lxc-attach -n "$CTID" -- bash -c "su docmost -c 'cd /opt/docmost && export NODE_OPTIONS=--max-old-space-size=4096 && pnpm install --frozen-lockfile && pnpm build'"

**Comment by rlkollman** [comment]
Per a recent PULL, this has been implemented, so closing. Thank you @MickLesk for adding Docmost!
https://github.com/community-scripts/ProxmoxVE/pull/2472
